src/Microsoft.Owin.Security.Facebook/FacebookAuthenticationHandler.cs

```diff
@@ -79,26 +80,63 @@ protected override async Task<AuthenticationTicket> AuthenticateCoreAsync()
                 string requestPrefix = Request.Scheme + "://" + Request.Host;
                 string redirectUri = requestPrefix + Request.PathBase + Options.CallbackPath;
 
-                string tokenRequest = "grant_type=authorization_code" +
+                string tokenRequestParams = "grant_type=authorization_code" +
                     "&code=" + Uri.EscapeDataString(code) +
                     "&redirect_uri=" + Uri.EscapeDataString(redirectUri) +
                     "&client_id=" + Uri.EscapeDataString(Options.AppId) +
                     "&client_secret=" + Uri.EscapeDataString(Options.AppSecret);
 
-                HttpResponseMessage tokenResponse = await _httpClient.GetAsync(Options.TokenEndpoint + "?" + tokenRequest, Request.CallCancelled);
+                var tokenRequest = new HttpRequestMessage(HttpMethod.Get, Options.TokenEndpoint + "?" + tokenRequestParams);
+                tokenRequest.Headers.Add("Accept", "application/json");
+                var tokenResponse = await _httpClient.SendAsync(tokenRequest, Request.CallCancelled);
                 tokenResponse.EnsureSuccessStatusCode();
                 string text = await tokenResponse.Content.ReadAsStringAsync();
-                JObject response = JObject.Parse(text);
 
-                string accessToken = response.Value<string>("access_token");
+                string accessToken, expires;
+                if (text.StartsWith("{"))
```

Comment:
Rather than sniffing the data, did you try checking the response content-type header?

Reply:
Content type is coming through as text/plain with the CGI style content body however I have no idea whether the JSON response is coming through as application/json because at the moment it is always behaving like this for me. I'm happy to revise the code but I'll need someone else to assist with testing the JSON based code-path.

Reply:
Since you have a reliable repro please contact Facebook's support so they get get to the bottom of this. Last I saw they still didn't understand why it was happening.

I don't want to check in a workaround for a temporary problem. We don't have a release scheduled so you'll need to build the package locally anyways. Build.cmd will produce the package you need, and you can adjust the version to 4.1.1-pre [here](https://github.com/aspnet/AspNetKatana/blob/e001fcf245ba4b6b2eb09ef410d398527882dc1a/build/Sakefile.shade#L4).

I will leave this PR open as a reference for anyone experiencing this issue. We can continue once there's a definitive answer from Facebook support.

Reply:
I completely agree with you about this PR - it's here only to help anyone experiencing the same issue however it certainly doesn't feel like something that should be a permanent fixture in the library!

I'm in conversation with Facebook support - so far the issue appears to be region specific as the token endpoint works (returning JSON) when driven through the Graph API explorer but does not when called by my app in the UK...